hooks/tk-multi-loader2/basic/scene_actions.py

```diff
@@ -157,16 +157,38 @@ def execute_action(self, name, params, sg_publish_data):
             raise IOError("File not found on disk - '%s'" % path)
 
         if name == _IMPORT:
-            self._import(path, is_sequence)
+            try:
+                self._import(path, is_sequence, sg_publish_data)
+            except Exception as e:
+                self.logger.exception("Unable to import %s: %s" % (path, e))
 
     ###########################################################################
     # helper methods
 
-    def _import(self, path, is_sequence):
+    def _import(self, path, is_sequence, sg_data):
         """
         Helper method to add the footage described by path to a comp
+
+        :param path: Path to the footage to add
+        :param is_sequence: Whether the footage is a sequence
+        :param sg_data: SG dictionary.
+        :returns:
         """
-        project = self.parent.engine.adobe.app.project
-        insertion_bin = project.getInsertionBin()
-        paths = [path.replace(os.path.sep, '/')]
-        project.importFiles(paths, 0, insertion_bin, is_sequence)
+        engine = self.parent.engine
+        current_project = engine.current_project
+        import_in_bin_template = self.parent.get_template("import_in_bin_template")
+        if import_in_bin_template:
+            import_bin_path = import_in_bin_template.apply_fields(sg_data)
+            self.logger.debug(
+                "Resolved import bin path to %s from %s with %s" % (import_bin_path, import_in_bin_template, sg_data))
+            current_bin = current_project.ensure_bins_for_path(import_bin_path)
+            current_bin.create_clip_from_media(path)
+        else:
+            # None is returned by get_template if the template can't be found, check if the setting was
+            # set to report the problem
+            template_setting = self.parent.get_setting("import_in_bin_template")
+            if template_setting:
+                raise ValueError("Invalid template setting %s" % template_setting)
+            insertion_bin = current_project.getInsertionBin()
+            paths = [path.replace(os.path.sep, '/')]
+            current_project.importFiles(paths, 0, insertion_bin, is_sequence)
```

Comment:
Please double check this works: if I'm not wrong `current_project` is an instance of our `PremiereProject` class. `importFiles` seems to be a call from the Adobe/Python api, so if you want to call it you should use `current_project.item.importFiles`. Same for `getInsertionBin`.